cli/cmd/project/logs.go

```diff
@@ -0,0 +1,109 @@
+package project
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rilldata/rill/cli/pkg/cmdutil"
+	"github.com/rilldata/rill/cli/pkg/config"
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	runtimeclient "github.com/rilldata/rill/runtime/client"
+	"github.com/spf13/cobra"
+)
+
+func LogsCmd(cfg *config.Config) *cobra.Command {
+	var name, path string
+	var follow bool
+	var tail int
+
+	logsCmd := &cobra.Command{
+		Use:   "logs [<project-name>]",
+		Args:  cobra.MaximumNArgs(1),
+		Short: "Show project logs",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := cmdutil.Client(cfg)
+			if err != nil {
+				return err
+			}
+			defer client.Close()
+
+			if len(args) > 0 {
+				name = args[0]
+			}
+
+			if !cmd.Flags().Changed("project") && len(args) == 0 && cfg.Interactive {
+				name, err = inferProjectName(cmd.Context(), client, cfg.Org, path)
+				if err != nil {
+					return err
+				}
+			}
+
+			proj, err := client.GetProject(context.Background(), &adminv1.GetProjectRequest{
+				OrganizationName: cfg.Org,
+				Name:             name,
+			})
+			if err != nil {
+				return err
+			}
+
+			depl := proj.ProdDeployment
+			if depl == nil {
+				return nil
+			}
+
+			if depl.Status != adminv1.DeploymentStatus_DEPLOYMENT_STATUS_OK {
+				cmdutil.PrintlnWarn(fmt.Sprintf("Deployment status not OK: %s", depl.Status.String()))
+				return nil
+			}
+
+			rt, err := runtimeclient.New(depl.RuntimeHost, proj.Jwt)
+			if err != nil {
+				return fmt.Errorf("failed to connect to runtime: %w", err)
+			}
+
+			if follow {
+				logClient, err := rt.WatchLogs(context.Background(), &runtimev1.WatchLogsRequest{InstanceId: depl.RuntimeInstanceId, Replay: true, ReplayLimit: int32(tail)})
+				if err != nil {
+					return fmt.Errorf("failed to watch logs: %w", err)
+				}
+
+				ctx, cancel := context.WithCancel(cmd.Context())
+
+				go func() {
+					for {
+						res, err := logClient.Recv()
+						if err != nil {
+							fmt.Println("failed to receive logs: %w", err)
+							cancel()
+							break
+						}
+						fmt.Println(res.Log)
```

Comment:
Maybe format the log nicely using a similar format as `rill start` (I think it's just a Zap development logger)

Reply:
you mean log this as json using logger created using `zap.NewDevelopment()`

Reply:
I mean to make it look like the format in `rill start`:

```
❯ rill start dev-project
2023-10-23T10:06:33.736	INFO	Serving Rill on: http://localhost:9009
2023-10-23T10:07:06.080	INFO	Reconciling resource	{"name": "metrics_margin_monitoring", "kind": "Source"}
2023-10-23T10:07:06.101	INFO	Ingesting source data	{"name": "metrics_margin_monitoring", "connector": "duckdb"}
2023-10-23T10:07:06.877	INFO	Reconciled resource	{"name": "metrics_margin_monitoring", "kind": "Source", "elapsed": "798ms"}
2023-10-23T10:07:06.877	INFO	Reconciling resource	{"name": "metrics_margin_model", "kind": "Model"}
2023-10-23T10:07:06.882	INFO	Reconciled resource	{"name": "metrics_margin_model", "kind": "Model", "elapsed": "5ms"}
2023-10-23T10:07:06.882	INFO	Reconciling resource	{"name": "customer_margin_dash", "kind": "MetricsView"}
2023-10-23T10:07:06.888	INFO	Reconciled resource	{"name": "customer_margin_dash", "kind": "MetricsView", "elapsed": "6ms"}
```